client/src/pages/ChatPage.js

```diff
@@ -0,0 +1,315 @@
+    setOpenName((cur) => !cur);
+  };
+
+  const onChangeName = (event) => {
+    dispatch(
+      renameChatAsync({
+        groupID: groupID,
+        name: name,
+      })
+    );
+    changeNameHandler();
+  };
+
+  const onInviteUser = (event) => {
+    setUserInvite(event.target.value);
+  };
+
+  const onExitUser = () => {
+    dispatch(
+      leaveChatAsync({
+        username: username,
+        groupID: groupID,
+      })
+    );
+    exitUserHandler();
+    navigate(`/studyGroups`);
+  };
+
+  const inviteDispatch = () => {
+    const updatedUser = {
+      username: userInvite,
+      groupID: groupID,
+    };
+
+    dispatch(inviteUserAsync(updatedUser));
+
+    addUserHandler();
+  };
+
+  if (!user.user) {
+    return null;
+  }
+
+  return (
+    <div>
+      <div className="fixed left-0 top-0 z-[1035] h-screen">
+        <SideBar />
+      </div>
+      <div className="p-5 !pl-[300px]">
+        <div className="flex justify-center">
+          <Chip
+            value={chat.name}
+            variant="outlined"
+            className="text-center text-indigo-300 w-3/5"
+          />
+        </div>
+
+        <div className="absolute top-0 right-0 m-5">
+          <IconButton
+            type="submit"
+            onClick={changeNameHandler}
+            className="flex bg-indigo-300 ml-5 mb-2"
+          >
+            <PencilSquareIcon className="h-5 w-5 " />
+          </IconButton>
+
+          <IconButton
+            type="submit"
+            onClick={addUserHandler}
+            className="flex bg-indigo-300 ml-5 mb-2"
+          >
+            <UserPlusIcon className="h-5 w-5" />
+          </IconButton>
+
+          <IconButton
+            type="submit"
+            onClick={exitUserHandler}
+            className="flex bg-indigo-300 ml-5 mb-2"
+          >
+            <ArrowRightOnRectangleIcon className="h-5 w-5" />
+          </IconButton>
+        </div>
+
+        <div className="flex justify-center">
+          <div className="mb-5 absolute bottom-0 w-3/5">
+            <Card className="flex mb-5">
+              <ul className="p-5 max-h-[30rem] overflow-y-auto scrollbar-width:none">
+                {chat.history
+                  ? chat.history.map((message, index) => (
+                      <li key={index}>
+                        <span className="font-bold">{message.username}: </span>
+                        {message.message}
+                      </li>
+                    ))
+                  : null}
+              </ul>
+            </Card>
+            <Card className="flex">
+              <form onSubmit={handleSubmit} className="flex p-5">
+                <Textarea
+                  label="Message"
+                  value={message}
+                  onChange={(event) => setMessage(event.target.value)}
+                />
+
+                <div className="flex items-center">
+                  <IconButton type="submit" className="flex bg-indigo-300 ml-5">
```

Comment:
Pressing enter does not send the chat which was unexpected

Reply:
I guess that was intentional but can def improve on it in the future